File: Repeated String/repeated-string.go
package main

import (
	"fmt"
	"strings"
)

func repeatedString(s string, n int64) int64 {
	if strings.Compare(s, "a") == 0 {
		return n
	} else if strings.Count(s, "a") == 0 {
		return 0
	}

	return (int64(strings.Count(s, "a")) * (n / int64(len(s)))) +
		int64(strings.Count(s[0:n % int64(len(s))], "a"))
}

func main() {
	strgs := []string{"abcac", "a", "ababa"}
	numbs := []int64{10, 1000000000000, 3}

	fmt.Printf("repeatedString(\"%v\", %v) = %d\n", strgs[0], numbs[0], repeatedString(strgs[0], numbs[0])) // 4
	fmt.Printf("repeatedString(\"%v\", %v) = %d\n", strgs[1], numbs[1], repeatedString(strgs[1], numbs[1])) // 1000000000000
	fmt.Printf("repeatedString(\"%v\", %v) = %d\n", strgs[2], numbs[2], repeatedString(strgs[2], numbs[2])) // 2
}

File: Repeated String/repeated-string.js
"use strict";

/**
 * Count repeats 'a' char in string
 *
 * @param {string} str
 * @return {int}
 */
let counter = str => {
  return str.split('').reduce((total, letter) => {
    return letter === 'a' ? total + 1 : total;
  }, 0);
};

/**
 * Print the number of letter a's in the first letters of infinite string
 *
 * @param {string} s
 * @param {int} n
 * @return {int}
 */
const repeatedString = (s, n) => {
  if (s === 'a') {
    return n;
  } else if (s.search('a') === -1) {
    return 0;
  }

  return (counter(s) * Math.floor(n / s.length)) + counter(s.substr(0, n % s.length));
};

console.log(repeatedString('abcac', 10)); // 4
console.log(repeatedString('aba', 10)); // 7
console.log(repeatedString('a', 1000000000000)); // 1000000000000